### PR TITLE
Baremetal Node overview

### PR DESCRIPTION
**Pull request by rawagner**
Depends on https://github.com/openshift/console/pull/4911

### CODE REVIEW

File: frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostDetails.tsx [hunk 1]
```diff
@@ -140,7 +140,7 @@ const BareMetalHostDetails: React.FC<BareMetalHostDetailsProps> = ({
           <dl>
             <dt>Status</dt>
             <dd>
-              <BareMetalHostStatus {...status} />
+              <BareMetalHostStatus {...status} maintenance={nodeMaintenance} host={host} />
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostDetails.tsx:143]: Is this change necessary? The `{...status}` already includes  maintenance or host. See `host-status.ts`

File: frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostStatus.tsx [hunk 2]
```diff
@@ -64,4 +70,8 @@ const BareMetalHostStatus: React.FC<StatusProps> = ({ status, title, description
   }
 };
 
+type BareMetalHostStatusProps = StatusProps & {
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostStatus.tsx:73]: I think we should keep the StatusProps in tact and specify it further here. The host should be optional as it is not necessary in all states

File: frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostsTable.tsx [hunk 3]
```diff
@@ -93,7 +93,7 @@ const HostsTableRow: RowFunction<BareMetalHostBundle> = ({
         />
       </TableData>
       <TableData className={tableColumnClasses.status}>
-        <BareMetalHostStatus {...status} />
+        <BareMetalHostStatus {...status} maintenance={nodeMaintenance} host={host} />
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostsTable.tsx:96]: maintenance and (or) host are provided through `{...status}`

File: frontend/packages/metal3-plugin/src/components/types.ts [hunk 4]
```diff
@@ -10,7 +10,8 @@ export type StatusProps = {
   status: string;
   title?: string;
   description?: string;
-  [key: string]: any;
+  maintenance?: K8sResourceKind;
+  className?: string;
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/types.ts:14]: I wouldn't add these two, this defines general shape of the status where it has status, title, description and a set of optional props which are provided as needed. These can be further specified in 'BareMetalHostStatusProps' and 'BareMetalNodeStatusProps'

File: frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostStatus.tsx (hunk 2 above)
Comment (rawagner) [reply frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostStatus.tsx:74]: OK, but I removed the `[key: string]: any` from `StatusProps` which basically removes any type-checking :) and added optional `nodeMaintenance` to both BMN and BMH statuses

Comment (jtomasek) [reply frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostStatus.tsx:74]: hmm, the goal of that was that status object has status, title, description and any amount of additional data which can be used by the status components so it does not have to be passed there explicitly and it is already included in the status object. With this in place, you can just do <BareMetalHostStatus {...status} /> without explicitly passing host and maintenance objects because they are already included in status object.

File: frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostsTable.tsx [hunk 5]
```diff
@@ -93,7 +93,7 @@ const HostsTableRow: RowFunction<BareMetalHostBundle> = ({
         />
       </TableData>
       <TableData className={tableColumnClasses.status}>
-        <BareMetalHostStatus {...status} />
+        <BareMetalHostStatus {...status} nodeMaintenance={nodeMaintenance} host={host} />
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostsTable.tsx:96]: nodeMaintenance and host are already included in status object, so no need to pass them explicitly here.

File: frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostDetails.tsx [hunk 6]
```diff
@@ -140,7 +140,7 @@ const BareMetalHostDetails: React.FC<BareMetalHostDetailsProps> = ({
           <dl>
             <dt>Status</dt>
             <dd>
-              <BareMetalHostStatus {...status} />
+              <BareMetalHostStatus {...status} nodeMaintenance={nodeMaintenance} host={host} />
```
Comment (jtomasek) [inline frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostDetails.tsx:143]: no need to add these as they are already included in status object
